src/manip_control/scripts/motion_interpolation.py

```diff
@@ -0,0 +1,108 @@
+import math
+
+
+class InterpolationSettings:
+
+    def __init__(self, acceleration, max_velocity, max_error, weights):
+        self.acceleration = acceleration
+        self.max_velocity = max_velocity
+        self.max_error = max_error
+        self.weights = weights
+
+    def from_params(params):
```

Comment:
a type annotation would be appreciated

Reply:
Turns out python doesn't support annotating type as the class you are defining (without cluttering the code with ugly workarounds). For that reason I chose to add annotation only to method argument.